Sources/DCoreKit/Model/NftModel.swift

```diff
@@ -0,0 +1,130 @@
+    case let property as NftProperty<UInt8>: return property.modifiableBy != .nobody
+    case let property as NftProperty<UInt32>: return property.modifiableBy != .nobody
+    case let property as NftProperty<UInt64>: return property.modifiableBy != .nobody
+    case let property as NftProperty<Bool>: return property.modifiableBy != .nobody
+    default: return false
+    }
+}
+
+extension NftModel {
+    func values() throws -> [AnyValue] {
+        return try Mirror(reflecting: self).children
+            .map { $0.value }
+            .compactMap(convertToAnyValue)
+    }
+
+    func createUpdate() throws -> [Pair<String, AnyValue>] {
+        return try Mirror(reflecting: self).children
+            .filter { isPropertyModifiable($0.value) }
+            .compactMap { property -> (String, AnyValue)? in
+                guard let name = property.label, let anyValue = try convertToAnyValue(property.value)
+                else { return nil }
+                return (name.dropFirst_(), anyValue)
+            }
+        .map { Pair($0.0, $0.1) }
+    }
+
+    func createDefinitions() -> [NftDataType] {
+        return Mirror(reflecting: self).children.map { child in
+            switch child.value {
+            case let property as NftProperty<String>: return property.asNftDataType(name: child.label)
+            case let property as NftProperty<Int>: return property.asNftDataType(name: child.label)
+            case let property as NftProperty<UInt8>: return property.asNftDataType(name: child.label)
+            case let property as NftProperty<UInt32>: return property.asNftDataType(name: child.label)
+            case let property as NftProperty<UInt64>: return property.asNftDataType(name: child.label)
+            case let property as NftProperty<Bool>: return property.asNftDataType(name: child.label)
+            default:
+                return NftDataType(
+                    type: getTypeLiteral(value: child.value), unique: false, modifiable: .nobody, name: child.label
+                )
+            }
+        }
+    }
+}
+
+private func getTypeLiteral(value: Any) -> NftDataType.TypeLiteral {
+    switch value {
+    case is Int, is UInt8, is UInt32, is UInt64: return .integer
+    case is Bool: return .boolean
+    case is String: return .string
+    default: preconditionFailure("Unsupported type \(value.self)")
+    }
+}
+
+private extension NftProperty {
+    func asNftDataType(name: String?) -> NftDataType {
+        return NftDataType(
+            type: getTypeLiteral(value: wrappedValue),
+            unique: unique,
+            modifiable: modifiableBy,
+            name: name?.dropFirst_()
+        )
+    }
+}
+
+/**
+ RawNft data model that only contains values of properties.
+ This model should be converted to a specific `NftModel` that contains specification for each property
+ */
+public struct RawNft: NftModel {
+    public let values: [AnyValue]
```

Comment:
values could not be public let values: AnyValue, cause AnyValue has an array case?

Reply:
it could in theory, but `values` will ALWAYS be array, so I'd have to use `if case .array()` on `values`, which would be less convenient.